src/category_theory/sites/sheafification.lean

```diff
@@ -621,6 +621,16 @@ adjunction.mk_of_hom_equiv
   end,
   hom_equiv_naturality_right' := λ P Q R η γ, by { dsimp, rw category.assoc } }
 
+instance Sheaf_to_presheaf_preserves_mono : (Sheaf_to_presheaf J D).preserves_monomorphisms :=
+functor.preserves_monomorphisms_of_adjunction (sheafification_adjunction _ _)
```

Comment:
I think that providing the right adjoint instance as @erdOne suggested is the better approach.
The rest looks good to me.